### PR TITLE
LPS-176874 Generate UAD module for BookmarksEntry, AnnouncementsEntry, and AnnouncementsFlag

### PR DESCRIPTION
**Pull request by julschong**
@tinatian @pei-jung
Hi Tina, PeiJung,
Per our discussion yesterday,
I was having issue generating for AnnouncementsFlag yesterday.
PeiJung suggested to add the uad attribute to the userId and it seems to have worked

> \<column name="userId" type="long" uad-anonymize-field-name="userId" /\>

### CODE REVIEW

**Review by pei-jung — APPROVED** [review]
LGTM! Thanks @julschong!

**Comment by pei-jung** [comment]
FYI, the [service builder](https://github.com/liferay/liferay-portal/blob/master/modules/util/portal-tools-service-builder/src/main/java/com/liferay/portal/tools/service/builder/ServiceBuilder.java#L931-L932) will only create the UAD classes if the entity is UAD-enabled, and an entity is considered UAD-enabled if one of it's columns is marked with the UAD attributes. Therefore, we need to add `uad-anonymize-field-name` to the `userId` column since it's the only UAD relevant field in AnnouncementsEntry.

**Comment by catarineaguiar** [comment]
no regressions found.

please, rename this [commit](https://github.com/liferay-core-infra/liferay-portal/pull/3692/commits/11ff834b28992fa200e032ee83e0b2e9f44e3a38) message with a reference to a required JIRA project and re-run sf.

Thanks,
Cat

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
@tinatian Hi Tina, 
would you please help take a look and submit this if no more issue?
Thanks!

File: portal-impl/src/com/liferay/portlet/announcements/service.xml [hunk 1]
```diff
@@ -106,7 +106,7 @@
 		<!-- Audit fields -->
 
 		<column name="companyId" type="long" />
-		<column name="userId" type="long" />
+		<column name="userId" type="long" uad-anonymize-field-name="userId" />
```
Comment (tinatian) [inline portal-impl/src/com/liferay/portlet/announcements/service.xml:109]: @julschong why this is changed? No generated code related to this change.

Comment (julschong) [reply portal-impl/src/com/liferay/portlet/announcements/service.xml:109]: @tinatian Hi Tina, This is related to BaseAnnouncementsFlagUADAnonymizer.

Comment (julschong) [reply portal-impl/src/com/liferay/portlet/announcements/service.xml:109]: If this line is not added, BaseAnnouncementsFlagUADAnonymizer does not generate at all

File: modules/apps/bookmarks/bookmarks-service/service.xml [hunk 2]
```diff
@@ -17,22 +17,22 @@
...
-		<column name="description" type="String" />
+		<column name="name" type="String" uad-nonanonymizable="true" />
+		<column name="url" type="String" uad-nonanonymizable="true" />
+		<column name="description" type="String" uad-nonanonymizable="true" />
 		<column name="priority" type="int" />
 		<column name="lastPublishDate" type="Date" />
 		<column name="status" type="int" />
 		<column name="statusByUserId" type="long" />
-		<column name="statusByUserName" type="String" />
+		<column name="statusByUserName" type="String" uad-anonymize-field-name="fullName" />
```
Comment (tinatian) [inline modules/apps/bookmarks/bookmarks-service/service.xml:35]: @julschong Line 20 and Line35 is adding same value `uad-anonymize-field-name="fullName"`, this does not look right to me

Comment (julschong) [reply modules/apps/bookmarks/bookmarks-service/service.xml:35]: @tinatian Hi Tina, I think you are right here. I see the the original non generated code is anonymizing using `anonymousUser.getScreenName()` I will check again and resend

**Pull request closed without merging**